flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/hybrid/HsFileDataIndexImplTest.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.hybrid;
+
+import org.apache.flink.runtime.io.network.partition.hybrid.HsFileDataIndex.ReadableRegion;
+import org.apache.flink.runtime.io.network.partition.hybrid.HsFileDataIndex.SpilledBuffer;
+import org.apache.flink.util.TestLoggerExtension;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** Tests for {@link HsFileDataIndexImpl}. */
+@ExtendWith(TestLoggerExtension.class)
+class HsFileDataIndexImplTest {
+    private static final int NUM_SUBPARTITIONS = 3;
+
+    private HsFileDataIndexImpl hsDataIndex;
+
+    @BeforeEach
+    void before() {
+        hsDataIndex = new HsFileDataIndexImpl(NUM_SUBPARTITIONS);
+    }
+
+    /**
+     * If a subpartition has no spilled buffer added to the index, all the query results are {@link
+     * Optional#empty()}.
+     */
+    @Test
+    void testGetReadableRegionForEmptySubpartition() {
+        // Only add buffers for subpartition 0.
+        hsDataIndex.addBuffers(createSpilledBuffers(0, Arrays.asList(0, 10)));
+        hsDataIndex.markBufferReadable(0, 0);
+        hsDataIndex.markBufferReadable(0, 10);
+        // Different subpartitions are independent and will not affect each other.
+        assertThat(hsDataIndex.getReadableRegion(1, 0)).isNotPresent();
+        assertThat(hsDataIndex.getReadableRegion(2, 10)).isNotPresent();
```

Comment:
Shouldn't this be good enough to server the purpose of this test case?
```suggestion
        assertThat(hsDataIndex.getReadableRegion(0, 0)).isNotPresent();
```